### ISSUE TITLE
Fix S1144: Should create issues for unused event handlers

### ISSUE THREAD

**Issue by nnpcYvIVl** [SonarSource/sonar-dotnet#2168]
S1144 flags M1 and M2 but not M3. Is this a false negative?

```csharp
namespace S1144_False_Negative
{
    class Class
    {
        private void M1() { }

        private void M2(object sender) { }

        private void M3(object sender, System.EventArgs e) { }
    }
}
```

### Related information

* SonarC# Version: Latest
* Visual Studio Version: 15.9.4

**Comment by valhristov** [comment]
The rule does not raise for event handlers because they could be attached in another file, for example in XAML. I see this is not documented, I created another ticket to update the documentation: #2169

**Comment by nnpcYvIVl** [comment]
> The rule does not raise for event handlers because they could be attached in another file, for example in XAML.

@valhristov Doesn't that only apply to partial classes? Could M3 in the example above be referenced in another file without without the partial declaration on the class?

**Comment by valhristov** [comment]
The classes are partial sometimes (always?), but still the event handler is often not referenced in the "other" part and we cannot detect if it is used or not. For example in WPF it is not.

**Comment by valhristov** [comment]
The partial type definition looks like a good clue that the event handler is declared in a class that is code-behind of a WPF control or something else. I'll make a test to see if any of the projects we analyze will generate false issues.

**Comment by nnpcYvIVl** [comment]
@valhristov Thanks for looking into this!
